web/modules/custom/solution/solution.module

```diff
@@ -15,3 +17,38 @@ function solution_entity_bundle_field_info_alter(&$fields, EntityTypeInterface $
     $fields['label']->addConstraint('UniqueSolutionTitle', []);
   }
 }
+
+/**
+ * Implements hook_entity_insert().
+ *
+ * The solution is connected to the collection with the affiliation field in the
+ * collection entity. The process of linking a solution to a collection is:
+ *  - Request an affiliation. This request is registered by adding the
+ *    collection to the affiliation_request field of the solution.
```

Comment:
Pff that doesn't seem like a nice workflow to me. That is really unclear for the end user. We should have a separate action for this.